### ISSUE TITLE
Intermittent work / fail behavior of OCP CAD Viewer

### ISSUE THREAD

**Issue by WayneSherman** [bernhard-42/vscode-ocp-cad-viewer#110]
Using
- a recently updated Manjaro (arch linux based)
- tested with both python 3.12.5 and python 3.10.14
- vscode 1.93.1
- Installed extensions
  code --install-extension ms-python.python
  code --install-extension bernhard-42.ocp-cad-viewer  # ver 2.4.1
- Used the "Quick build123d" install button

After following the installation instructions, when opening the auto created demo file "ocp_vscode_demo.py" the viewer pane intermittently does not initialize properly.

It behaves like an initialization race condition.  Changing the way I load the file changes the timing and the failure frequency, e.g. opening a folder (workspace) which contains the demo file vs opening the demo file directly.  When opening a folder the viewer fails to initialize most of the time, whereas when opening the file directly the viewer pane works most of the time.  But both cases have intermittent work/fail behavior.

My test system is on the slow side (running in VirtualBox) so it probably is exposing this race condition where a faster system would not see it.

_Originally posted by @WayneSherman in https://github.com/bernhard-42/vscode-ocp-cad-viewer/discussions/104_

**Comment by bernhard-42** [comment]
I found one problem that could be related to it. The viewer tries to respond to a resize window event and call `viewer.resize` when `viewer` is not ready initialized. Hence the viewer crashes and does not work properly any more. 
On fast machines, the viewer might be initialized before the resize event hits. The fix is to check whether `viewer` is not `null` before responding on the resize event.
But, I did not check this against a "slow system" - due to not having a slow system :-)

**Comment by bernhard-42** [comment]
I've published v2.5.0 which fixes the potential race condition. Could you please test it since I can't reproduce it

**Comment by CePeU** [comment]
I testet a fresh install (windows OS) and the hanging empty viewer window on first start remained. I will make a seperate issue in the following days.

**Comment by bernhard-42** [comment]
@CePeU Yes please, since there was no issue open I forgot to deal with it

**Comment by WayneSherman** [comment]
> I've published v2.5.0 which fixes the potential race condition. Could you please test it, @WayneSherman, since I can't reproduce it

Thanks for the updates.   I tested v2.5.0 and the behavior is unchanged.  When I "Open Folder..." with the demo file folder, most often the viewer window does not display properly.  When I "Open File..." with the demo file, the viewer window usually displays correctly.

**Comment by bernhard-42** [comment]
I have no idea how to reproduce this

**Comment by bernhard-42** [comment]
Closing for now
